COMETwebapp/ViewModels/Components/Shared/ParameterEditors/BooleanParameterTypeEditorViewModel.cs

```diff
@@ -42,7 +42,7 @@ public class BooleanParameterTypeEditorViewModel : ParameterTypeEditorBaseViewMo
         /// <param name="parameterType">the parameter used for this editor view model</param>
         /// <param name="valueSet">the value set asociated to this editor</param>
         /// <param name="isReadOnly">The readonly state</param>
-        public BooleanParameterTypeEditorViewModel(BooleanParameterType parameterType, IValueSet valueSet, bool isReadOnly) : base(parameterType, valueSet, isReadOnly)
+        public BooleanParameterTypeEditorViewModel(BooleanParameterType parameterType, IValueSet valueSet, bool isReadOnly, int compoundIndex = -1) : base(parameterType, valueSet, isReadOnly, compoundIndex)

```

Comment:
General comment: change the compoundIndex to valueArrayIndex with defaut value of 0. The OnParameterValueChanged method could then be modify as : 
```                   
modifiedValueArray = new ValueArray<string>(this.CurrentParameterSwitchKind)
{
    [valueArrayIndex] = valueString
};
```

For component, also you the valueArrayIndex to have the value